### PR TITLE
refactor(user-notification): Refactor user notification to use the config object in services. Set emailFromAddress in config.

### PR DESCRIPTION
**Pull request by saevarma**
## What

* Refactor user-notification to user the config object in services and remove custom property providers.
* Add emailFromAddress to config.
* Add global queue setup to prevent concurrent race condition between test files on queue creation.

## Why

To simplify app configuration and setup, this is cleaner.
To make email from address configurable as there are different sender emails whitelisted in AWS.

## Screenshots / Gifs

N/A

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Updated email sender addresses for different environments to ensure consistent email delivery.
  - Integrated new client modules and feature flags for improved functionality and user experience in notification services.

- **Bug Fixes**
  - Adjusted import orders and configurations to resolve potential issues with service initialization and dependencies.

- **Refactor**
  - Streamlined the codebase by removing redundant imports and declarations.
  - Improved organization and maintainability by restructuring environment settings.

- **Chores**
  - Enhanced test setups for notification services with new configurations and dependencies for comprehensive testing environments.

### CODE REVIEW

**Review by rafnarnason — APPROVED** [review]
lgtm

File: apps/services/user-notification/test/globalSetup.ts [hunk 1]
```diff
@@ -20,6 +23,27 @@ const startSQS = async () => {
...
+    },
+  })
+
+  await client.send(
+    new CreateQueueCommand({ QueueName: environment.MAIN_QUEUE_NAME }),
+  )
+  await client.send(
+    new CreateQueueCommand({ QueueName: environment.DEAD_LETTER_QUEUE_NAME }),
+  )
+}
```
[the remark this answers is not in the document]
Comment (saevarma) [reply apps/services/user-notification/test/globalSetup.ts:42]: This fails jest test startup immediately and jest logs the error in terminal, so this should suffice without explicit try/catch

[the remark this answers is not in the document]
Comment (saevarma) [reply apps/services/user-notification/test/globalSetup.ts]: I pushed this change to add explicit log line to aid developers in recognising this type of test failure.

File: apps/services/user-notification/infra/user-notification.ts [hunk 2]
```diff
@@ -39,12 +39,12 @@ const getEnv = (services: {
   AUTH_DELEGATION_MACHINE_CLIENT_SCOPE: json([
     '@island.is/auth/delegations/index:system',
   ]),
-  USER_NOTIFICATION_APP_PROTOCOL: {
```
Comment (busla) [inline apps/services/user-notification/infra/user-notification.ts:42]: wont this break the `env-switcher` support in the test app?

**Review by busla — CHANGES_REQUESTED** [review]
I think we might break the test app

**Comment by saevarma** [comment]
> I think we might break the test app

Can you elaborate more on that?

File: apps/services/user-notification/infra/user-notification.ts (hunk 2 above)
Comment (saevarma) [reply apps/services/user-notification/infra/user-notification.ts:42]: This was not being used anywhere in the notification service. The app registered a device token, so I assume that this is some legacy code as the test app is just registered with device token in dev environment.

Comment (busla) [reply apps/services/user-notification/infra/user-notification.ts:42]: this is part of the callback url schema in Cognito that we use for the env switcher in the test app.

Callback configuration in Cognito

```
is.island.app.dev://cognito
is.island.app.staging://cognito
is.island.app://cognito
```

Comment (busla) [reply apps/services/user-notification/infra/user-notification.ts:42]: will be fixed with deeplinks, resolving.

**Review by busla — APPROVED** [review]
env-switcher will break but is being fixed with universal links

 @saevarma @thoreyjona

**Pull request merged**